File: examples/swagger-3/src/test/resources/asserts/java/spring_cloud_openfeign/src/com/wuxp/codegen/swagger3/clients/OrderFeignClient.java
package com.wuxp.codegen.swagger3.clients;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;
import org.springframework.http.MediaType;

      import com.wuxp.codegen.swagger3.domain.Order;
      import java.util.Set;
      import com.wuxp.codegen.swagger3.evt.QueryOrderEvt;
      import com.wuxp.codegen.swagger3.evt.CreateOrderEvt;
      import com.wuxp.codegen.swagger3.resp.ExampleDTO;
      import com.wuxp.codegen.swagger3.resp.PageInfo;
      import com.wuxp.codegen.swagger3.resp.ServiceResponse;
      import java.util.List;
      import java.util.Map;

    /**
     * 接口：GET
    **/

  @FeignClient(
        path = "/order" 
  )
public interface OrderFeignClient{

    /**
      * 1:GET /order/getOrder
      * 2:Http请求方法：GET
      * 3:返回值在java中的类型为：List
      * 4:返回值在java中的类型为：Order
     **/
      @GetMapping()
    List<Order>  getOrder (
          @RequestParam(name = "names" )  String[] names,
          @RequestHeader(name = "My-Ids" )  List<Integer> ids,
  Set<Order> moneys
  );
    /**
      * 1:GET /order
      * 2:Http请求方法：GET
      * 3:返回值在java中的类型为：PageInfo
      * 4:返回值在java中的类型为：Order
     **/
      @GetMapping()
    PageInfo<Order>  queryOrder2 (
  QueryOrderEvt evt
  );
    /**
      * 1:POST /order
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：PageInfo
      * 4:返回值在java中的类型为：Order
     **/
      @PostMapping()
    PageInfo<Order>  queryOrder (
          @RequestBody()  QueryOrderEvt evt
  );
    /**
      * 1:POST /order/queryOrder3
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：PageInfo
      * 4:返回值在java中的类型为：Order
     **/
      @PostMapping()
    PageInfo<Order>  queryOrder3 (
          @RequestBody()  QueryOrderEvt[] evt
  );
    /**
      * 1:POST /order/queryOrder4
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：PageInfo
      * 4:返回值在java中的类型为：Order
     **/
      @PostMapping()
    PageInfo<Order>  queryOrder4 (
          @RequestBody()  Set<QueryOrderEvt> evt
  );
    /**
      * 1:POST /order/queryOrder5
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：PageInfo
      * 4:返回值在java中的类型为：Order
     **/
      @PostMapping()
    PageInfo<Order>  queryOrder5 (
          @RequestBody()  Map<String,QueryOrderEvt> evt
  );
    /**
      * 1:POST /order/queryOrder6
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：PageInfo
      * 4:返回值在java中的类型为：Order
     **/
      @PostMapping()
    PageInfo<Order>  queryOrder6 (
          @RequestBody()  List<QueryOrderEvt> evt
  );
    /**
      * 1:POST /order/queryOrder_2
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：ServiceQueryResponse
      * 4:返回值在java中的类型为：Order
     **/
      @PostMapping(produces = {[HttpMediaType.MULTIPART_FORM_DATA]} )
    ServiceResponse<PageInfo<Order>>  queryOrder_2 (
          @RequestParam(name = "order_id" ,required = false )  Long oderId,
  String sn
  );
    /**
      * 1:POST /order/queryPage
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：ServiceResponse
      * 4:返回值在java中的类型为：PageInfo
      * 5:返回值在java中的类型为：Order
     **/
      @PostMapping()
    ServiceResponse<PageInfo<Order>>  queryPage (
  String id
  );
    /**
      * 1:GET /order/createOrder
      * 2:Http请求方法：GET
      * 3:返回值在java中的类型为：ServiceResponse
      * 4:返回值在java中的类型为：Long
     **/
      @GetMapping()
    ServiceResponse<Long>  createOrder (
  CreateOrderEvt evt
  );
    /**
      * 1:POST /order/hello
      * 2:Http请求方法：POST
      * 3:返回值在java中的类型为：ServiceResponse
      * 4:返回值在java中的类型为：Object
     **/
      @PostMapping()
    ServiceResponse<Object>  hello (
  );
    /**
      * 1:DELETE /order/delete
      * 2:Http请求方法：DELETE
      * 3:返回值在java中的类型为：ServiceResponse
      * 4:返回值在java中的类型为：Object
     **/
      @DeleteMapping(value = "/delete" )
    ServiceResponse<Object>  delete (
  ExampleDTO dto
  );
}

File: examples/swagger-3/codegen-sdk/swagger3/dart/dart_feign/src/example/services/order_feign_client.dart
import 'dart:io';
import 'package:built_collection/built_collection.dart';
import 'package:built_value/built_value.dart';
import 'package:built_value/serializer.dart';
import 'package:fengwuxp_dart_basic/index.dart';
import 'package:fengwuxp_dart_openfeign/index.dart';

          import '../../domain/order.dart';
          import '../../resp/page_info.dart';
          import '../../evt/create_order_evt.dart';
          import '../../evt/query_order_evt.dart';
          import '../../evt/example_dto.dart';
          import '../../serializers.dart';


      /// 接口：GET
@Feign
  @FeignClient(value:"/order",)
class OrderFeignClient extends FeignProxyClient {

OrderFeignClient() : super() {

}


      /// 1:Http请求方法：GET
      /// 2:返回值在java中的类型为：List
      /// 3:返回值在java中的类型为：Order
          @GetMapping()
  Future<BuiltList<Order>>  getOrder(
          @RequestParam(name: "names" )
        BuiltList<String> names,
          @RequestHeader(name: "ids" )
        BuiltList<int> ids,
        BuiltSet<Order> moneys,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<BuiltList<Order>>("getOrder",
  [names,ids,moneys,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
              specifiedType:FullType(BuiltList,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：GET
      /// 2:返回值在java中的类型为：PageInfo
      /// 3:返回值在java中的类型为：Order
          @GetMapping()
  Future<PageInfo<Order>>  queryOrder2(
        QueryOrderEvt evt,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryOrder2",
  [evt,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：PageInfo
      /// 3:返回值在java中的类型为：Order
          @PostMapping()
  Future<PageInfo<Order>>  queryOrder(
          @RequestBody()
        QueryOrderEvt evt,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryOrder",
  [evt,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：PageInfo
      /// 3:返回值在java中的类型为：Order
          @PostMapping()
  Future<PageInfo<Order>>  queryOrder3(
          @RequestBody()
        BuiltList<QueryOrderEvt> evt,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryOrder3",
  [evt,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：PageInfo
      /// 3:返回值在java中的类型为：Order
          @PostMapping()
  Future<PageInfo<Order>>  queryOrder4(
          @RequestBody()
        BuiltSet<QueryOrderEvt> evt,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryOrder4",
  [evt,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：PageInfo
      /// 3:返回值在java中的类型为：Order
          @PostMapping()
  Future<PageInfo<Order>>  queryOrder5(
          @RequestBody()
        BuiltMap<String,QueryOrderEvt> evt,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryOrder5",
  [evt,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：PageInfo
      /// 3:返回值在java中的类型为：Order
          @PostMapping()
  Future<PageInfo<Order>>  queryOrder6(
          @RequestBody()
        BuiltList<QueryOrderEvt> evt,
        int memberId,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryOrder6",
  [evt,memberId,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：ServiceQueryResponse
      /// 3:返回值在java中的类型为：Order
          @PostMapping(produces:[HttpMediaType.MULTIPART_FORM_DATA],)
  Future<PageInfo<Order>>  queryOrder_2(
          @RequestParam(name: "order_id" ,required: false )
        int oderId,
        String sn,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryOrder_2",
  [oderId,sn,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：ServiceResponse
      /// 3:返回值在java中的类型为：PageInfo
      /// 4:返回值在java中的类型为：Order
          @PostMapping()
  Future<PageInfo<Order>>  queryPage(
        String id,
        int memberId,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<PageInfo<Order>>("queryPage",
  [id,memberId,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
                serializeType: PageInfo,
              specifiedType:FullType(PageInfo,[FullType(Order)])
          )
    
  );
  }

      /// 1:Http请求方法：GET
      /// 2:返回值在java中的类型为：ServiceResponse
      /// 3:返回值在java中的类型为：Long
          @GetMapping()
  Future<int>  createOrder(
        CreateOrderEvt evt,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<int>("createOrder",
  [evt,],
          feignOptions: feignOptions,
          serializer: BuiltValueSerializable(
              specifiedType:FullType(int)
          )
    
  );
  }

      /// 1:Http请求方法：POST
      /// 2:返回值在java中的类型为：ServiceResponse
      /// 3:返回值在java中的类型为：Object
          @PostMapping()
  Future<Object>  hello(
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<Object>("hello",
  [],
    feignOptions: feignOptions
  );
  }

      /// 1:Http请求方法：DELETE
      /// 2:返回值在java中的类型为：ServiceResponse
      /// 3:返回值在java中的类型为：Object
          @DeleteMapping(value:"/delete",)
  Future<Object>  delete(
        ExampleDTO dto,
  [UIOptions? feignOptions]) {
  return this.delegateInvoke<Object>("delete",
  [dto,],
    feignOptions: feignOptions
  );
  }
}


final orderFeignClient = OrderFeignClient();
